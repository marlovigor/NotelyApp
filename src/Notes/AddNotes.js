import React from 'react'
import { Link } from 'react-router-dom'
import AppContext from '../AppContext'



const newDate = new Date()

class AddNotes extends React.Component{
	state = {
		"folderId": this.props.match.params.folderId,
		"id": Math.ceil(Math.random()*10000000).toString(),
		"modified": newDate,
		"name": '',
		"content": ''
	}
	static contextType = AppContext

	handleChange= event =>{
		const {name, value} = event.target
		this.setState({
			[name]: value
		})
	}
	handleSubmit=event=>{
		event.preventDefault()
		fetch('http://localhost:9090/notes', {
			method: 'POST',
			body: JSON.stringify(this.state),
			headers: {'content-type': 'application/json'}
		})
		.then(this.context.handleNoteSubmit())
	}
	handleClick=()=>{
		console.log(this.state)
	}


	render(){
		return(
			<div>
				<form onSubmit={this.handleSubmit}>
					<label htmlFor="noteName">New Note Name:</label>
					<br/>
					<input type="text" onChange={this.handleChange} className="noteName" name="name" required/>
					<br/>
					<label htmlFor="noteContent" >Note content:</label>
					<br/>
					<textarea name="content" onChange={this.handleChange} cols="100" rows="10" required></textarea>
					<br/>
					<button type="submit" onClick={this.handleClick}>Submit</button>
				</form>
			</div>
		)
	}
}

export default AddNotes