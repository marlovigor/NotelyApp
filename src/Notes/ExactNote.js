import React from 'react'
import AppContext from '../AppContext'
import { Link } from 'react-router-dom'

class ExactNote extends React.Component{
	static contextType = AppContext

	handleDelete = event => {
		event.preventDefault()
		const noteId = this.props.match.params.noteId
		fetch(`http://localhost:3000/notes/${noteId}`, {
      		method: 'DELETE',
      		headers: {'content-type': 'application/json'},
    		})
			.then(response => response.json())
			.then(() => this.context.deleteNote())
			.catch(error => console.log(error))
		}
	render(props){
		const exactNote = this.context.notes.find(note => note.id === this.props.match.params.noteId)
		return(
			<div>
				<h2>{exactNote.name}</h2>
				<p>{exactNote.content}</p>
				<Link to='/' ><button onClick={() => this.context.deleteNote(this.props.match.params.noteId)}>Delete</button></Link>
			</div>
		)
	}
}
export default ExactNote