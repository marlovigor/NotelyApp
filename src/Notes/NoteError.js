import React from 'react'
import GoBack from '../Folders/GoBack'

class NoteError extends React.Component{
	state = {
		hasError: false
	}

	static getDerivedStateFromError(error){
		return {hasError: true}
	}

	render(){
		return (
			this.state.hasError ? <h2>There was an error with your notes <GoBack /></h2> : this.props.children
			)
	}
}

export default NoteError