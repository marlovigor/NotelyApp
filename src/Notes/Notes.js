import React from 'react'
import { Link } from 'react-router-dom'
import AppContext from '../AppContext'

class Notes extends React.Component{
	static contextType = AppContext
	render(props){
		return (
			<div className="notes">
				<Link to={`/Notes/${this.props.id}`} ><button className="note">{this.props.name}</button></Link>
			</div>
		)}
}

export default Notes