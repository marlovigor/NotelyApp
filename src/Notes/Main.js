import React from 'react'
import {Link} from 'react-router-dom'
import Notes from './Notes'

export default function Main(props){
	const notes = props.notes.map(note => <li><Notes id={note.id} name={note.name} content={note.content} /></li>)
	return(
		<div>
			<ul>
				{notes}
			</ul>
			<button>Add Note</button>
		</div>)
}