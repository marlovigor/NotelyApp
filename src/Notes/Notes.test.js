import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import Notes from './Notes';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<BrowserRouter><Notes id="1" /></BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
const tree = renderer
  .create(<BrowserRouter><Notes id="1" name="dolphins"/></BrowserRouter>)
  .toJSON();
expect(tree).toMatchSnapshot();  
});