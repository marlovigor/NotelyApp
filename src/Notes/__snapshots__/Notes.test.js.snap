// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="notes"
>
  <a
    href="/Notes/1"
    onClick={[Function]}
  >
    <button
      className="note"
    >
      dolphins
    </button>
  </a>
</div>
`;
